apis/ops/v1alpha1/perconaxtradb_ops_types.go

```diff
@@ -64,43 +66,90 @@ type PerconaXtraDBOpsRequestSpec struct {
 	// +kubebuilder:default="IfReady"
 	Apply ApplyOption `json:"apply,omitempty"`
 }
 
-// PerconaXtraDBReplicaReadinessCriteria is the criteria for checking readiness of a PerconaXtraDB pod
-// after updating, horizontal scaling etc.
-type PerconaXtraDBReplicaReadinessCriteria struct{}
-
 type PerconaXtraDBUpgradeSpec struct {
 	// Specifies the target version name from catalog
-	TargetVersion     string                                 `json:"targetVersion,omitempty"`
-	ReadinessCriteria *PerconaXtraDBReplicaReadinessCriteria `json:"readinessCriteria,omitempty"`
+	TargetVersion string `json:"targetVersion,omitempty"`
 }
 
-// HorizontalScaling is the spec for PerconaXtraDB horizontal scaling
-type PerconaXtraDBHorizontalScalingSpec struct{}
+type PerconaXtraDBHorizontalScalingSpec struct {
+	// Number of nodes/members of the group
+	Member *int32 `json:"member,omitempty"`
+	// specifies the weight of the current member/Node
+	MemberWeight int32 `json:"memberWeight,omitempty"`
+}
 
-// PerconaXtraDBVerticalScalingSpec is the spec for PerconaXtraDB vertical scaling
 type PerconaXtraDBVerticalScalingSpec struct {
-	ReadinessCriteria *PerconaXtraDBReplicaReadinessCriteria `json:"readinessCriteria,omitempty"`
+	PerconaXtraDB *core.ResourceRequirements `json:"perconaxtradb,omitempty"`
+	Exporter      *core.ResourceRequirements `json:"exporter,omitempty"`
+	Coordinator   *core.ResourceRequirements `json:"coordinator,omitempty"`
 }
 
 // PerconaXtraDBVolumeExpansionSpec is the spec for PerconaXtraDB volume expansion
-type PerconaXtraDBVolumeExpansionSpec struct{}
+type PerconaXtraDBVolumeExpansionSpec struct {
+	PerconaXtraDB *resource.Quantity   `json:"perconaxtradb,omitempty"`
+	Mode          *VolumeExpansionMode `json:"mode,omitempty"`
+}
 
-type PerconaXtraDBCustomConfigurationSpec struct{}
+type PerconaXtraDBCustomConfigurationSpec struct {
+	// ConfigSecret is an optional field to provide custom configuration file for database.
+	// +optional
+	ConfigSecret *core.LocalObjectReference `json:"configSecret,omitempty"`
+	// Deprecated
+	InlineConfig string `json:"inlineConfig,omitempty"`
+	// If set to "true", the user provided configuration will be removed.
+	// PerconaXtraDB will start will default configuration that is generated by the operator.
+	// +optional
+	RemoveCustomConfig bool `json:"removeCustomConfig,omitempty"`
+	// ApplyConfig is an optional field to provide PerconaXtraDB configuration.
+	// Provided configuration will be applied to config files stored in ConfigSecret.
+	// If the ConfigSecret is missing, the operator will create a new k8s secret by the
+	// following naming convention: {db-name}-user-config .
+	// Expected input format:
+	//	applyConfig:
+	//		file-name.cnf: |
+	//			[mysqld]
+	//			key1: value1
+	//			key2: value2
+	// +optional
+	ApplyConfig map[string]string `json:"applyConfig,omitempty"`
+}
 
 type PerconaXtraDBCustomConfiguration struct {
```

Comment:
didn't find any usage

Reply:
Here is the usage:
https://github.com/kubedb/ops-manager/blob/add-px-ops/pkg/controller/perconaxtradb/reconfigure.go#L68-L164